src/Raven.Server/Documents/Document.cs

```diff
@@ -102,37 +104,55 @@ public bool Expired(DateTime currentDate)
             return false;
         }
 
-        private static bool IsMetadataEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument)
+        private static DocumentCompareResult IsMetadataEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument, bool skipAttachment)
         {
             if (targetDocument == null)
-                return false;
+                return DocumentCompareResult.DifferenceDetected;
 
             BlittableJsonReaderObject myMetadata;
             BlittableJsonReaderObject objMetadata;
             currentDocument.TryGet(Constants.Documents.Metadata.Key, out myMetadata);
             targetDocument.TryGet(Constants.Documents.Metadata.Key, out objMetadata);
 
             if (myMetadata == null && objMetadata == null)
-                return true;
+                return DocumentCompareResult.Equal;
 
-            if (myMetadata == null || objMetadata == null)
-                return false;
+            if (myMetadata == null)
+                return DocumentCompareResult.Equal;
 
-            return ComparePropertiesExceptionStartingWithAt(myMetadata, objMetadata, isMetadata: true);
+            if (objMetadata == null)
+            {
+                targetDocument.Modifications = new DynamicJsonValue(targetDocument)
+                {
+                    [Constants.Documents.Metadata.Key] = myMetadata
+                };
+                return DocumentCompareResult.Equal | DocumentCompareResult.ShouldRecreateDocument;
+            }
+
+            return ComparePropertiesExceptionStartingWithAt(myMetadata, objMetadata, true, skipAttachment);
         }
 
-        public static bool IsEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument)
+        public static DocumentCompareResult IsEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument, 
```

Comment:
Same here. Side-effect. Also shouldn't this be a part of some ConflictResolver or something like that?